core/modules/views/plugins/views_plugin_cache.inc

```diff
@@ -284,60 +284,78 @@ class views_plugin_cache extends views_plugin {
     }
   }
 
-  function get_results_key() {
-    global $user;
-
+  public function get_results_key() {
     if (!isset($this->_results_key)) {
-
-      $build_info = $this->view->build_info;
-
-      $query_plugin = $this->view->display_handler->get_plugin('query');
-
-      foreach (array('query','count_query') as $index) {
-        // If the default query back-end is used generate SQL query strings from
-        // the query objects.
-        if ($build_info[$index] instanceof SelectQueryInterface) {
-          $query = clone $build_info[$index];
-          $query->preExecute();
-          $build_info[$index] = (string) $query;
-        }
-      }
-      $key_data = array(
-        'build_info' => $build_info,
-        'roles' => array_keys($user->roles),
-        'super-user' => $user->uid == 1, // special caching for super user.
-        'language' => $GLOBALS['language']->langcode,
-        'base_url' => $GLOBALS['base_url'],
-      );
+      $key_data = array();
       foreach (array('exposed_info', 'page', 'sort', 'order', 'items_per_page', 'offset') as $key) {
         if (isset($_GET[$key])) {
           $key_data[$key] = $_GET[$key];
         }
       }
 
-      $this->_results_key = $this->view->name . ':' . $this->display->id . ':results:' . md5(serialize($key_data));
+      $this->_results_key = $this->view->name . ':' . $this->display->id . ':results:' . $this->get_cache_key($key_data);
     }
 
     return $this->_results_key;
   }
 
-  function get_output_key() {
-    global $user;
+  public function get_output_key() {
     if (!isset($this->_output_key)) {
       $key_data = array(
         'result' => $this->view->result,
-        'roles' => array_keys($user->roles),
-        'super-user' => $user->uid == 1, // special caching for super user.
         'theme' => $GLOBALS['theme'],
-        'language' => $GLOBALS['language']->langcode,
-        'base_url' => $GLOBALS['base_url'],
       );
-
-      $this->_output_key = $this->view->name . ':' . $this->display->id . ':output:' . md5(serialize($key_data));
+      $this->_output_key = $this->view->name . ':' . $this->display->id . ':output:' . $this->get_cache_key($key_data);
     }
 
     return $this->_output_key;
   }
+
+  /**
+   * Returns cache key.
+   *
+   * @param array $key_data
+   *   Additional data for cache segmentation and/or overrides for default
+   *   segmentation.
+   *
+   * @return string
+   */
+  public function get_cache_key($key_data = array()) {
+    global $user;
+
+    $key_data += array(
+      'roles' => array_keys($user->roles),
+      'super-user' => $user->uid == 1,
+    // special caching for super user.
+      'language' => $GLOBALS['language']->langcode,
+      'language_content' => $GLOBALS['language_content']->langcode,
+      'base_url' => $GLOBALS['base_url'],
```

Comment:
Thanks for including the changes from https://git.drupalcode.org/project/views/-/commit/aff92c08791e5bb01271e5e661375d04f46fcd8a in the line prior to this one @jromine 🙏🏼  ...can we please also include the line from https://git.drupalcode.org/project/views/-/commit/d5703d76b4ca079cd64ae1a64f7b62e779b50d69?:

```suggestion
      'base_url' => $GLOBALS['base_url'],
      `'display_options' => $this->display->display_options,`
```